packages/core/test/model.test.ts

```diff
@@ -58,6 +63,9 @@ describe('createModel', () => {
                   age: e.value
                 };
               })
+            },
+            anotherEvent: {
+              actions: clearName
```

Comment:
Removing `exec` property (a contravariant property) from here:
https://github.com/statelyai/xstate/blob/c6ead65609a591bc6949e90d572ff43932dfc4a4/packages/core/src/types.ts#L44
fixes the problem

However, then we stop inferring the event type correctly here:
https://github.com/statelyai/xstate/blob/2ab14af07b5ef4b07279d729086dedcd1ea352d1/packages/core/test/model.test.ts#L61

It's not being narrowed down to a specific event and instead `e` becomes a union of all events.